From 0bb6e03dbbf0bd47f6f8cc42274b8f7fa9fc9262 Mon Sep 17 00:00:00 2001
From: sinanmohd <sinan@sinanmohd.com>
Date: Sat, 17 Feb 2024 14:31:03 +0530
Subject: [PATCH] common: fix config location when wrapped

---
 bin/common | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/bin/common b/bin/common
index 7ac7977..941e77a 100644
--- a/bin/common
+++ b/bin/common
@@ -332,6 +332,9 @@ search_config() {
 # Read config file
 read_config() {
   local name=$(echo "$1" | sed -e 's/[^A-Za-z0-9._-]/_/g')
+  name="${1%-wrapped.conf}"
+  name="${name#.}.conf"
+
   local config=$(search_config "$name")
   if [ "$config" ]; then
     . "$config"
-- 
2.43.0

