From f0cefc2567b17127fef861c4491eb8e97ec61c7f Mon Sep 17 00:00:00 2001
From: sinanmohd <sinan@sinanmohd.com>
Date: Tue, 12 May 2026 22:46:13 +0530
Subject: [PATCH] fix: align toolbar text

---
 src/main.rs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/main.rs b/src/main.rs
index 4828923..88f4ecb 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -230,7 +230,7 @@ fn main() {
             .unwrap_or("?");
 
         tooltip += &format!(
-            "{} {} {} {} {} {}%\n",
+            "{} {} {} {} {} {}%\n\n",
             if args.nerd { "󰖜" } else { "🌅" },
             format_ampm_time(day, "sunrise", args.ampm),
             if args.nerd { "󰖛" } else { "🌇" },
@@ -255,7 +255,7 @@ fn main() {
 
             let sep = if args.tabs { "\t" } else { " " };
             let mut tooltip_line = format!(
-                "{}{sep}{}{sep}{}{sep}{}",
+                "{}  \t{}  \t{}  \t{}",
                 format_time(hour["time"].as_str().unwrap(), args.ampm),
                 if args.nerd {
                     WEATHER_CODES_NERD
-- 
2.54.0

